tests/providers/snowflake/decorators/test_snowpark.py

```diff
@@ -0,0 +1,168 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import unittest
+from unittest import mock
+
+from airflow.decorators import task
+from airflow.models.dag import DAG
+from airflow.utils import timezone
+
+try:
+    import snowflake.snowpark as snowpark
+    from snowflake.snowpark import Session
+except ImportError:
+    snowpark = None
+
+DEFAULT_DATE = timezone.datetime(2021, 9, 1)
+
+
+class TestDockerDecorator:
+    @unittest.skipIf(snowpark is None, 'snowpark package not present')
```

Comment:
Shouldn't we apply this decorator to class? In this way we don't have to duplicate the code.

Reply:
As far as I know, the decorators in the `unittest` package cannot be used for classes.

Reply:
You can use `pytest.mark.skipif` instead, which does work on classes.